### PR TITLE
Script Design Audit - store all actions

### PR DESCRIPTION
**Pull request by Suyash-d**
**Description**
We want to store all the actions performed on script design information by end users when performed via the REST API. All this information is stored in a dedicated table AUD_SCRIPT. The information stored is the following
* ID: 
  * description: a random UUID
  * type: string (255)
  * nullable: false
  * primary key
* USERNAME: 
  * description: username of the user
  * type: string (255)
  * nullable: false
* USER_ID: 
  * description: uuid of the user
  * type: string (255)
  * nullable: false
* ACTION: 
  * description: enumeration of the possible actions of the users:
    * CREATE
    * UPDATE
    * DELETED
  * type: string (255)
  * nullable: false
* TIMESTAMP: 
  * description: timestamp of the action
  * type: string (255)
  * nullable: false
* SCRIPT_ID: 
  * description: id of the script being updated
  * type: string (255)
  * nullable: false
* SCRIPT_NM: 
  * description: name of the script being updated
  * type: string (2000)
  * nullable: false
* SCRIPT_VERSION:
  * description: version of the script being updated
  * type: number
  * nullable: false
* SECURITY_GROUP: 
  * description: security group of the script being updated
  * type: string (255)
  * nullable: false

**Tasks**
iesi-core project:
1. update application-metadata.yml & src/main/resources/application-iesi-metadata.yml (don't forget core/conf/application-metadata.yml)
2. create a new ScriptDesignAuditKey class under a new package io/metadew/iesi/metadata/definition/audit that extends MetadataKey and that uses a UUID as primary key (see io.metadew.iesi.metadata.definition.user.UserKey for inspiration)
3. create a new ScriptDesignAudit class under a new package io/metadew/iesi/metadata/definition/audit that extends Metadata (see io.metadew.iesi.metadata.definition.user.User for inspiration)
4. create a new ScriptDesignConfiguration class under a new package io/metadew/iesi/metadata/configuration/audit that extends the Configuration class (see io.metadew.iesi.metadata.configuration.user.UserConfiguration for inspiration)
5. Create an enum ScriptDesignAuditAction with the different options: CREATE, UPDATE, DELETE

iesi-rest project
1. add ScriptDesignAuditConfiguration to the IesiConfiguration as a bean (see io/metadew/iesi/server/rest/configuration/IesiConfiguration.java:104 for inspiration)
2. for every Put/Post/DeleteMapping in ScriptsController, add the necessary audit step to the method: ScriptDesignAuditConfiguration#insert(...) where the username & username or fetched from the current SecurityContextHolder (see io/metadew/iesi/server/rest/executionrequest/ExecutionRequestController.java:122)

### CODE REVIEW

File: core/java/iesi-rest-without-microservices/src/test/resources/application-iesi-metadata.yml [hunk 1]
```diff
@@ -2805,6 +2805,67 @@ iesi:
...
+            order: 8
+            type: NUMBER
+            length: 0
+            nullable: false
+          SECURITY_GROUP:
+            description: security group of the script being updated
+            order: 9
+            type: STRING
+            length: 255
+            nullable: false
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/test/resources/application-iesi-metadata.yml:2868]: remove empty file 'git' here below

File: core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/audit/key/ScriptDesignAuditKey.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.util.UUID;
+
+@Data
+@AllArgsConstructor
+@Builder
+@EqualsAndHashCode(callSuper = true)
+public class ScriptDesignAuditKey extends MetadataKey {
+
+    private final UUID ID;
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/definition/audit/key/ScriptDesignAuditKey.java:17]: change parameter name to lowercase

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/audit/IScriptDesignAuditPostDtoService.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package io.metadew.iesi.server.rest.script.audit;
+
+import io.metadew.iesi.metadata.definition.audit.ScriptDesignAudit;
+import io.metadew.iesi.metadata.definition.audit.ScriptDesignAuditAction;
+import io.metadew.iesi.metadata.definition.script.Script;
+
+public interface IScriptDesignAuditPostDtoService {
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/audit/IScriptDesignAuditPostDtoService.java:7]: rename to IScriptDesignAuditService

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/audit/ScriptDesignAuditPostDtoService.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import io.metadew.iesi.server.rest.user.UserDtoRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Service
+public class ScriptDesignAuditPostDtoService implements IScriptDesignAuditPostDtoService{
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/script/audit/ScriptDesignAuditPostDtoService.java:17]: rename to ScriptDesignAuditService

**Comment by BerrevoetsRobbe** [comment]
@Suyash-d I'm unable to delete a script from the UI using this branch. Please verify this using the sandbox built by CircleCI

File: core/java/iesi-rest-without-microservices/src/test/resources/application-iesi-metadata.yml (hunk 1 above)
Comment (Suyash-d) [reply core/java/iesi-rest-without-microservices/src/test/resources/application-iesi-metadata.yml:2868]: Removed

**Review by BerrevoetsRobbe — COMMENTED** [review]
approved

**Comment by BerrevoetsRobbe** [comment]
please add the alter/create table script

**Comment by BerrevoetsRobbe** [comment]
@Suyash-d please fix merge conflicts

File: core/java/iesi-rest-without-microservices/src/main/resources/application-iesi-metadata.yml [hunk 5]
```diff
@@ -2572,13 +2572,74 @@ iesi:
             length: 0
             nullable: true
             defaultTimestamp:
-          ACTION_TYP_NM:
+           ACTION_TYP_NM:
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/main/resources/application-iesi-metadata.yml:2575]: indent?

**Pull request merged**